src/Simplify_Select.cpp

```diff
@@ -21,18 +21,25 @@ Expr Simplify::visit(const Select *op, ExprInfo *info) {
 
         // clang-format off
         if (EVAL_IN_LAMBDA
-            (rewrite(select(IRMatcher::likely(true), x, y), x) ||
-             rewrite(select(IRMatcher::likely(false), x, y), y) ||
-             rewrite(select(IRMatcher::likely_if_innermost(true), x, y), x) ||
-             rewrite(select(IRMatcher::likely_if_innermost(false), x, y), y) ||
-             rewrite(select(1, x, y), x) ||
-             rewrite(select(0, x, y), y) ||
-             rewrite(select(x, y, y), y) ||
+            (rewrite(select(IRMatcher::likely(true), x, y), true_value) ||
```

Comment:
are these changes meant to be part of this PR? What's going on here?

Reply:
Running the fuzzer with debugging turned on identified an issue here where information wasn't being propagated through a constant-folded select as aggressively as it could have been. It happened to have a bitwise op in it, which is why I was looking at the case and puzzling over the too-loose bounds.